schema/migration-2-0001-20211003.sql

```diff
@@ -88,7 +88,6 @@ BEGIN
     EXECUTE 'ALTER TABLE "tx_metadata" ADD CONSTRAINT "unique_tx_metadata" UNIQUE("key","tx_id")' ;
     EXECUTE 'ALTER TABLE "tx_metadata" ADD CONSTRAINT "tx_metadata_tx_id_fkey" FOREIGN KEY("tx_id") REFERENCES "tx"("id") ON DELETE CASCADE  ON UPDATE RESTRICT' ;
     EXECUTE 'CREATe TABLE "reward"("id" SERIAL8  PRIMARY KEY UNIQUE,"addr_id" INT8 NOT NULL,"type" rewardtype NOT NULL,"amount" lovelace NOT NULL,"earned_epoch" INT8 NOT NULL,"spendable_epoch" INT8 NOT NULL,"pool_id" INT8 NULL)' ;
-    EXECUTE 'ALTER TABLE "reward" ADD CONSTRAINT "unique_reward" UNIQUE("addr_id","type","earned_epoch","pool_id")' ;
```

Comment:
This will need to be a new migration file which drops the unique key rather than deleting this line. This would allow us to have proper migration, so that users don't need to resync from genesis when upgrading. To create this new file, you could check https://github.com/input-output-hk/cardano-db-sync/blob/master/doc/schema-management.md#creating-a-migration as well as https://github.com/input-output-hk/cardano-db-sync/blob/master/doc/migrations.md for general context about migrations.

Reply:
That's what I thought we did, but I must of got mixed up when we talked about it as I made a note that line should be delete. 😅

Reply:
Actualy it may be possible to just delete it, but it needs more special attention.

If there is an upgrade from an existing `13.1.x.x` version and the Unique key is already there this won't cause any issues. What could cause issues is using the next release schema with an existing `13.1.x.x` release. This can happen on a downgrade or when someone uses snapshots created after the change with older release.

So bottom line, yes you could leave it as is, but the unique key should be created when a snapshot is created, in the `runPrepareSnapshot` function.